src/distributions/dirichlet.py

```diff
@@ -1,63 +1,60 @@
 import numpy as np
 import matplotlib.pyplot as plt
-from scipy.stats import dirichlet
-
-def save_to(directory: str, extension: str):
-    def plot_dirichlet(alpha, ax):
-        """
-        Plots a Dirichlet distribution given alpha parameters and axis.
-        """
-        # Create a 2D meshgrid of points
-        resolution = 200  # Resolution of the visualization
-        x = np.linspace(0, 1, resolution)
-        y = np.linspace(0, 1, resolution)
-        X, Y = np.meshgrid(x, y)
-
-        # Calculate remaining coordinate for the 3-simplex (3D Dirichlet is defined on a triangle in 2D)
-        Z = 1 - X - Y
-
-        # Filter out points outside the triangle
-        valid = Z >= 0
-
-        # Prepare coordinates and the probability density function (PDF) array
-        X = X[valid]
-        Y = Y[valid]
-        Z = Z[valid]
-        points = np.vstack((X, Y, Z)).T  # Stack the valid points
-
-        # Calculate the Dirichlet PDF for valid points
-        PDF = dirichlet.pdf(points, alpha)
-
-        # Reshape PDF back into a 2D array for contour plotting
-        # Initialize full grid PDF to zero (areas outside the triangle remain zero)
-        full_PDF = np.zeros((resolution, resolution))
-        full_PDF[valid] = PDF  # Place the calculated PDF in the valid grid positions
-
-        # Create a contour plot on the provided axis
-        contour = ax.contourf(np.linspace(0, 1, resolution), np.linspace(0, 1, resolution), full_PDF, levels=50, cmap='Blues')
-        plt.colorbar(contour, ax=ax, pad=0.05, aspect=10)
+import matplotlib.tri as tri
+from math import gamma
+
+
+# Code source: https://blog.bogatron.net/blog/2014/02/02/visualizing-dirichlet-distributions/
+class Dirichlet(object):
+    def __init__(self, alpha):
+        self._alpha = np.array(alpha)
+        self._coef = gamma(np.sum(self._alpha)) / np.multiply.reduce([gamma(a) for a in self._alpha])
+
+    def pdf(self, x):
+        """Returns pdf value for `x`."""
+        return self._coef * np.multiply.reduce([xx ** (aa - 1) for (xx, aa) in zip(x, self._alpha)])
+
+
+def save_to(directory: str, _: str):
+    extension = "png"  # Hardcode png output format. SVG output for Dirichlet is ~22 MB, while png is ~115KB.
+    corners = np.array([[0, 0], [1, 0], [0.5, 0.75 ** 0.5]])
```

Comment:
I notice that your plots use the same parameters as on Wikipedia but with a different coordinate layout. This is fine, but it would be nice to have some sort of labelling of axes on the plots.

Reply:
I agree, I haven't been able to figure out how to achieve that with matplotlib yet. I'll keep researching